cmd/server/cadence/cadence.go

```diff
@@ -111,34 +114,72 @@ func BuildCLI(releaseVersion string, gitRevision string) *cli.App {
 				},
 			},
 			Action: func(c *cli.Context) error {
-				fxApp := fx.New(
-					config.Module,
-					logfx.Module,
-					dynamicconfigfx.Module,
-					fx.Provide(func() appContext {
-						return appContext{
-							CfgContext: config.Context{
-								Environment: getEnvironment(c),
-								Zone:        getZone(c),
-							},
-							ConfigDir: getConfigDir(c),
-							RootDir:   getRootDir(c),
-							Services:  getServices(c),
-						}
-					}),
-					Module,
-				)
+				stoppedWg := &sync.WaitGroup{}
 
-				ctx := context.Background()
-				if err := fxApp.Start(ctx); err != nil {
-					return err
-				}
+				ctx, cancel := context.WithCancel(context.Background())
+				defer cancel()
 
-				// Block until FX receives a shutdown signal
-				<-fxApp.Done()
+				services := getServices(c)
+				// non blocking channel that can accommodate at least 3 errors per service (worst case).
+				errChan := make(chan error, len(services)*3)
```

Comment:
I suggest using a blocking channel and printing errors as soon as possible. Currently if we fail to start a single service we will start the others and only print the error after the others complete